pkg/monitor/cluster/debugpods.go

```diff
@@ -0,0 +1,69 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Each cluster is being checked every 1 minute. emitDebugPodsCount
+// lists all events (age < 70 seconds) in the default namespace
+// in order to detect recent events of debug-pod creation.
+func (mon *Monitor) emitDebugPodsCount(ctx context.Context) error {
+	events, err := mon.cli.CoreV1().Events("default").List(ctx, metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(map[string]string{
+			"reason":              "Started",
+			"involvedObject.kind": "Pod",
+			// oc debug node creates a pod with a "container-00"
+			// container name.
+			"involvedObject.fieldPath": "spec.containers{container-00}",
+		}).String(),
+	})
+	if err != nil {
+		return err
+	}
+
+	nodes, err := mon.cli.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	podNames := getDebugPodNames(nodes.Items)
+
+	count := countDebugPods(podNames, events.Items)
+	mon.emitGauge("debugpods.count", int64(count), nil)
+
+	return nil
+}
+
+func countDebugPods(debugPods []string, events []corev1.Event) int {
+	count := 0
+	for _, e := range events {
+		if eventIsNew(e) && stringutils.Contains(debugPods, e.InvolvedObject.Name) {
+			count++
+		}
+	}
+	return count
+}
+
+func eventIsNew(event corev1.Event) bool {
+	// corev1.Event.LastTimeStamp is deprecated in newer k8s releases.
+	// Update this check when ARO's k8s version is upgraded
+	return time.Since(event.LastTimestamp.Time) < time.Second*70
+}
+
+func getDebugPodNames(nodes []corev1.Node) []string {
+	names := make([]string, len(nodes))
+	for i, n := range nodes {
+		names[i] = fmt.Sprintf("%s-debug", n.Name)
```

Comment:
I believe this has a low chance to not work in case the node names are weird. If you check how the debug pod name is created in source code for `oc debug`, it does this:
`fmt.Sprintf("%s-debug", generateapp.MakeSimpleName(infos[0].Name))` (see
https://github.com/openshift/oc/blob/master/pkg/cli/debug/debug.go#L446) with `MakeSimpleName` stripping some characters. Could we use the same helperfunction? https://github.com/openshift/oc/blob/f94afb52dc8a3185b3b9eacaf92ec34d80f8708d/pkg/helpers/newapp/app/pipeline.go#L327

Reply:
maybe the chance isn't even that low, because it also lowercases the name